### DISCUSSION TITLE
JS Scripting in automation

### DISCUSSION THREAD

**Discussion by AndyGUK** [Budibase/budibase#4628] [Help] [answered]
I'm hoping that someone with a bit more knowledge of JS and Budibase may be able to help me understand how I can access items within the `{{ trigger.fields }}` object.

My automation will follow three basic steps: 
1. When this happens: APP ACTION
2. JS Scripting to understand change made by user 
3. Create new row in MySQL

However, being new to JS I'm finding it to setup the JS successfully. Ultimately, my question is: how do I reference the individual fields within the object that Budibase creates when an action happens in the app?

At the moment, my code looks like this:
```
function gads_cust_check (){
//Check if Google Ads Customer STATUS has changed
if (fields.ORGINAL_gads_cust_status !== fields.NEW_gads_cust_status) {
  const gads_cust_status_change = `The Google Ads customer status was changed to ${fields.NEW_gads_cust_status}`;
   return gads_cust_status_change;
} 
```
Any help will be greatly appreciated!

**Comment by mjashanks** [comment, marked as answer]
Hey @AndyGUK - you're almost there!

your full code should be:

```javascript
//Check if Google Ads Customer STATUS has changed
if (trigger.fields.ORGINAL_gads_cust_status !== trigger.fields.NEW_gads_cust_status) {
  const gads_cust_status_change = `The Google Ads customer status was changed to ${trigger.fields.NEW_gads_cust_status}`;
  return gads_cust_status_change;
} else {
  return "No Change";
}
```

Points:
1. You don't need to wrap everything in a `function() {... }`
2. You always should `return` a value - I've added an `else` condition
3. You should use `trigger.fields...` instead of `fields...`

**Reply by AndyGUK** [reply]
Thank you SO much for this... feel like I've been going round in circles for the last three days! This works perfect, thanks again!
